### PR TITLE
Calculate bounding box for geojson queries

### PR DESCRIPTION
**Pull request by cameronmarsh**
https://github.com/smartcitiesdata/discovery_api/issues/188

### CODE REVIEW

File: lib/discovery_api_web/utilities/geojson_utils.ex [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def update_bounding_box(_, _), do: raise(MalformedGeometryError)
+
+  defp get_max(a, nil), do: a
+  defp get_max(a, b), do: max(a, b)
+
+  defp reduce_coordinates([x, y]) when is_list(x) == false and is_list(y) == false do
+    {x, y}
+  end
```
Comment (abrock-accenture) [inline lib/discovery_api_web/utilities/geojson_utils.ex]: Sweet. I was wondering how to solve the problem of "how do we know when to stop flattening these lists?"

File: test/unit/discovery_api_web/utilities/geojson_utils_test.exs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      where([
+        [:geometry_type, :coordinates, :expected_bounding_box],
+        ["point", [[1, 0]], [1, 0, 1, 0]],
+        ["line", [[1, 0], [2, 0], [2, 1]], [1, 0, 2, 1]],
+        [
+          "polygon",
+          [[[1, 1], [1, 0], [0, 1], [1, 1]], [[3, 3], [3, 2], [2, 3], [3, 3]]],
+          [0, 0, 3, 3]
+        ],
+        ["empty coordinates", [], nil]
```
Comment (abrock-accenture) [inline test/unit/discovery_api_web/utilities/geojson_utils_test.exs]: I think you cover all of the possible coordinate structures here, but check your understand against [this](https://en.wikipedia.org/wiki/GeoJSON) - we got burned on multi-line strings before.

File: lib/discovery_api_web/utilities/geojson_utils.ex [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      min(x, min_x),
+      min(y, min_y),
+      get_max(x, max_x),
+      get_max(y, max_y)
+    ]
+  end
+
+  def update_bounding_box(_, _), do: raise(MalformedGeometryError)
+
+  defp get_max(a, nil), do: a
```
Comment (ManApart) [inline lib/discovery_api_web/utilities/geojson_utils.ex]: It makes sense that this guy works 100% of the time with the current code, but it is sort of a temporal coupling (we assume the nil will ALWAYS be the second param). To save us from making a mistake as the code is updated in the future we should also add `defp get_max(nil, b), do: b`.

Comment (LtChae) [reply lib/discovery_api_web/utilities/geojson_utils.ex]: A nil in the source data is handled elsewhere. This head handles the initial nil value in the bounding_box accumulator.

Comment (ManApart) [reply lib/discovery_api_web/utilities/geojson_utils.ex]: Still seems safer to me to cover both cases here incase code is changed elsewhere and someone doesn't realize that this is temporally coupled to that source data being nil.

**Pull request merged**